olive/passes/olive_pass.py

```diff
@@ -74,6 +74,8 @@ def __init__(
         config_class, default_config = self.get_config_class(accelerator_spec, disable_search)
 
         self._accelerator_spec = accelerator_spec
+        self._is_accelerator_agnostic = True
```

Comment:
I think we could make this a class attribute like 
![image](https://github.com/microsoft/Olive/assets/94929125/6161ce65-160a-4930-bcea-42131724db2d)

Otherwise, I don't see where the pass developer sets this property without changing `__init__`. Or do we expect the value for this property to change based on the pass run config? I think it is simpler if the pass just has a class attribute for this?

Reply:
I deliberately not make it class attribute is because for some passes, it could/could not depend hardware by different EP. 
For example, the quantization could depend hardware if the EP is DML. it should be independent on hardware if the EP is CPU. In this way, the CPU quantization could be reused if the EP is CPU or other EP that doesn't impact the quantization behavior. The dml quantization should only be reused if the EP is DML. 

How do you think?